### PR TITLE
Remove .components-item-group selector in edit-site components

### LINKED ISSUE #67369: Remove usage of `.components-item-group` CSS selector from edit-site components
### Description

Splitting this out from https://github.com/WordPress/gutenberg/pull/67321#discussion_r1860202855

Cc @t-hamano @jameskoster 

CSS selectors that come from base components of the `@wordpress/components` package should not be used in other packages.

This is the only place in all of Gutenberg where the .components-item-group selector is used. 
It should be replaced with a more specific and 'semantically' correct CSS class. The class should be applied to all the `<ItemGroup>` components used in the edit site navigation screen and all its sub-screens,

Introduced in https://github.com/WordPress/gutenberg/pull/50767

https://github.com/WordPress/gutenberg/blob/b54d1fe5fe96b00d7f6455b4aaf40bb21ec43967/packages/edit-site/src/components/sidebar-navigation-screen/style.scss#L21-L24

### Step-by-step reproduction instructions

- Search in the codebase for usages of the `.components-item-group` CSS selector.
- Observe there's only one usage in a component of the edit-site package.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

### PR DESCRIPTION
**Pull request by im3dabasia**
Fixes: #67369 

## What?
The `.components-item-group` selector has been replaced with a custom class (`.edit-site-sidebar-navigation-screen__item-group`) that is scoped specifically to the sidebar navigation screen.

This change does not affect other components outside of the sidebar navigation screen and will prevent unintended side effects if the `ItemGroup` component is used elsewhere in the future.

## Screenshot

Screenshot 2024-11-28 at 2 38 10 PM

### CODE REVIEW

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR!

There are no strict rules, but in the Gutenberg project, many components define CSS class names based on the following rules:

```
{package-name}-{component-name|file-name}
```

This component is in the `edit-site` package and the file name is `sidebar-navigation-screen-main`, so an example of one CSS class name would be `edit-site-sidebar-navigation-screen-main`.

I don't think we should include the components used inside the component (in this case the `ItemGroup` component) in the CSS class name.

What do you think about these points?

**Comment by im3dabasia** [comment]
> What do you think about these points?

Yes, I saw how the class names are given in the project. I appreciate your example, and I think the name you suggested is appropriate for our context.

I will make the changes right away.

**Comment by im3dabasia** [comment]
I have made the changes as suggested. Please check now, @t-hamano. Let me know if there are any further changes.

**Comment by t-hamano** [comment]
Thanks for the update!

Sorry, I missed an important point. The `.edit-site-sidebar-navigation-screen__content .components-item` selector is applied to more than just the Design screen. We need to apply horizontal negative margins to these screens as well:

<details><summary>Navigation</summary>

https://github.com/WordPress/gutenberg/blob/52b5429f52096c5e080b3cc5a2ff696e61cb7616/packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/index.js#L119

image

</details> 

<details><summary>Pages</summary>

https://github.com/WordPress/gutenberg/blob/52b5429f52096c5e080b3cc5a2ff696e61cb7616/packages/edit-site/src/components/sidebar-dataviews/index.js#L29

image

</details> 

<details><summary>Templates</summary>

https://github.com/WordPress/gutenberg/blob/52b5429f52096c5e080b3cc5a2ff696e61cb7616/packages/edit-site/src/components/sidebar-navigation-screen-templates-browse/content.js#L60

image

</details> 

<details><summary>Patterns</summary>

For this screen, [the `edit-site-sidebar-navigation-screen-patterns__group` class name](https://github.com/WordPress/gutenberg/blob/52b5429f52096c5e080b3cc5a2ff696e61cb7616/packages/edit-site/src/components/sidebar-navigation-screen-patterns/index.js#L137) is already applied, so we can use it.

image

</details> 


Can I use a similar approach to add class names to these components? 🙏

**Comment by im3dabasia** [comment]
> Can I use a similar approach to add class names to these components?

Yes @t-hamano , I have done as you have suggested please check now. 

**Navigation Screen**
Picture 1  

**Pages Screen**
Picture 2  

**Templates Screen**
Picture 3  

**Patterns Screen**
Picture 4  


Thank you for reviewing my work @t-hamano . Please let me know if any changes are needed in the PR.

File: packages/edit-site/src/components/sidebar-dataviews/style.scss [hunk 1]
```diff
@@ -27,3 +27,8 @@
 		color: $white;
 	}
 }
+
+.edit-site-sidebar-dataviews {
+	margin-left: -$grid-unit-20;
+	margin-right: -$grid-unit-20;
+}
```
Comment (t-hamano) [inline packages/edit-site/src/components/sidebar-dataviews/style.scss]: The element with this CSS class is a parent of the element with the `.edit-site-sidebar-dataviews-dataview-item` class, so let's move this CSS [here](https://github.com/WordPress/gutenberg/pull/67384/files#diff-78833cb86b24921f76235fb7e680f682c4516dc18560ad67ea135ccfea85e3e2R9).

File: packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/style.scss [hunk 2]
```diff
@@ -46,3 +46,8 @@
 .edit-site-sidebar-navigation-screen-navigation-menus__helper-block-editor {
 	display: none;
 }
+
+.edit-site-sidebar-navigation-screen-navigation-menus {
+	margin-left: -$grid-unit-20;
+	margin-right: -$grid-unit-20;
+}
```
Comment (t-hamano) [inline packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/style.scss]: Let's move this CSS class [here](https://github.com/WordPress/gutenberg/pull/67384/files#diff-ed0c8c20650bc9a6acdd0da8c3e459af70b86b90a52a9550f628423b7b63b5d7R4) as well.

File: packages/edit-site/src/style.scss [hunk 3]
```diff
@@ -35,6 +35,7 @@
 @import "./components/pagination/style.scss";
 @import "./components/global-styles/variations/style.scss";
 @import "./components/sidebar-global-styles-wrapper/style.scss";
+@import "./components/sidebar-navigation-screen-templates-browse/style.scss";
```
Comment (t-hamano) [inline packages/edit-site/src/style.scss]: Let's group the CSS related to the navigation screen together. Moving it [here](https://github.com/WordPress/gutenberg/pull/67384/files#diff-5065a00f747fc9cdddcc2220e19e8e56df99e52f99cc25e1a4e51a21daf0897aR19) seems like a good place to start.

File: packages/edit-site/src/components/sidebar-navigation-screen/style.scss [hunk 4]
```diff
@@ -18,7 +18,7 @@
 .edit-site-sidebar-navigation-screen__content {
 	padding: 0 $grid-unit-20;
 
-	.components-item-group {
+	.edit-site-sidebar-navigation-screen-main {
```
Comment (t-hamano) [inline packages/edit-site/src/components/sidebar-navigation-screen/style.scss]: Can we create a new `packages/edit-site/src/components/sidebar-navigation-screen-main/style.scss` file and move this style into it?

**Review by t-hamano — COMMENTED** [review]
Thanks for the update!

Everything works as expected, but it would be nice to improve the code quality a bit more.

Additionally, the latest trunk resolves some issues with mobile layouts. To be on the safe side, it would be a good idea to rebase this PR and retest 🙏

**Comment by im3dabasia** [comment]
Hey @t-hamano, I’m really sorry for this. I think I made a mistake while doing the rebase. I’ll create a new PR and attach it to this issue. I'm really sorry for any inconvenience caused.

New PR Link : https://github.com/WordPress/gutenberg/pull/67575

**Pull request closed without merging**